examples/all-clusters-app/all-clusters-common/BUILD.gn

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2020 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build_overrides/chip.gni")
+
+config("includes") {
+  include_dirs = [
+    ".",
+    "include",
+  ]
+}
+
+source_set("all-clusters-common") {
+  sources = [
+    "${chip_root}/examples/common/chip-app-server/DataModelHandler.cpp",
+    "${chip_root}/src/app/clusters/barrier-control-server/barrier-control-server.cpp",
+    "${chip_root}/src/app/clusters/basic/basic.cpp",
+    "${chip_root}/src/app/clusters/color-control-server/color-control-server.cpp",
+    "${chip_root}/src/app/clusters/door-lock-server/door-lock-server-core.cpp",
+    "${chip_root}/src/app/clusters/door-lock-server/door-lock-server-logging.cpp",
+    "${chip_root}/src/app/clusters/door-lock-server/door-lock-server-schedule.cpp",
+    "${chip_root}/src/app/clusters/door-lock-server/door-lock-server-user.cpp",
+    "${chip_root}/src/app/clusters/groups-client/groups-client.cpp",
+    "${chip_root}/src/app/clusters/groups-server/groups-server.cpp",
+    "${chip_root}/src/app/clusters/ias-zone-server/ias-zone-server.cpp",
+    "${chip_root}/src/app/clusters/identify/identify.cpp",
+    "${chip_root}/src/app/clusters/level-control/level-control.cpp",
+    "${chip_root}/src/app/clusters/on-off-server/on-off.cpp",
+    "${chip_root}/src/app/clusters/scenes-client/scenes-client.cpp",
+    "${chip_root}/src/app/clusters/scenes/scenes.cpp",
+    "${chip_root}/src/app/clusters/temperature-measurement-server/temperature-measurement-server.cpp",
+    "${chip_root}/src/app/reporting/reporting-default-configuration.cpp",
+    "${chip_root}/src/app/reporting/reporting.cpp",
+    "${chip_root}/src/app/util/af-event.cpp",
+    "${chip_root}/src/app/util/af-main-common.cpp",
+    "${chip_root}/src/app/util/attribute-size.cpp",
+    "${chip_root}/src/app/util/attribute-storage.cpp",
+    "${chip_root}/src/app/util/attribute-table.cpp",
+    "${chip_root}/src/app/util/binding-table.cpp",
+    "${chip_root}/src/app/util/chip-message-send.cpp",
+    "${chip_root}/src/app/util/client-api.cpp",
+    "${chip_root}/src/app/util/ember-print.cpp",
+    "${chip_root}/src/app/util/message.cpp",
+    "${chip_root}/src/app/util/process-cluster-message.cpp",
+    "${chip_root}/src/app/util/process-global-message.cpp",
+    "${chip_root}/src/app/util/util.cpp",
+    "gen/call-command-handler.c",
+    "gen/callback-stub.c",
+    "gen/clusters-callback-stubs.c",
+  ]
+
```

Comment:
As a possible followup, worth checking whether adding `-ffunction-sections` to cflags and c++ flags fixes the linker issues....

Reply:
I opened #3718 to not forgot to look at that.